### ISSUE TITLE
Bug: When a department is specificed with include/exclude, the inherit mode is ignored on specific cops.

### ISSUE THREAD

**Issue by rhiroe** [rubocop/rubocop#13757]
When using RuboCop-1.71 with [rubocop-on-rbs](https://github.com/ksss/rubocop-on-rbs), some .rbs files started receiving warnings for `Style/FrozenStringLiteralComment`. Not all .rbs files receive warnings; only those .rbs files that can be parsed as Ruby code (e.g., files with just class definitions) are flagged. https://github.com/rubocop/rubocop/pull/13699#issuecomment-2615011523

Reproducible repository: https://github.com/rhiroe/hoge

According to @Earlopain, this is a bug in RuboCop, where the inherit mode is ignored on specific cops when a department is specified with include/exclude. Thank you for the investigation, @Earlopain !

ref: https://github.com/ksss/rubocop-on-rbs/pull/76#issuecomment-2616060233

---

## Expected behavior

Even when a department is specified with include/exclude, inherit_mode is not ignored.

## Actual behavior

When a department is specified with include/exclude, the inherit mode is ignored on specific cops.

## Steps to reproduce the problem

https://github.com/rhiroe/hoge

## RuboCop version

```
$ bundle exec rubocop -V
1.71.0 (using Parser 3.3.7.0, rubocop-ast 1.37.0, analyzing as Ruby 2.7, running on ruby 3.4.1) [aarch64-linux]
```

**Comment by Earlopain** [comment]
Simple Repo:

```yml
# .rubocop.yml
Style:
  Exclude:
    - foo.rb
```

```rb
# foo.rb
''
```

```bash
$ bundle exec rubocop foo.rb --only Style/FrozenStringLiteralComment

Inspecting 1 file
C

Offenses:

foo.rb:1:1: C: [Correctable] Style/FrozenStringLiteralComment: Missing frozen string literal comment.
''
^

1 file inspected, 1 offense detected, 1 offense autocorrectable
```

**Comment by Earlopain** [comment]
I think this is actually a dup of https://github.com/rubocop/rubocop/issues/9983

**Comment by rhiroe** [comment]
I agree. I will close this issue.
